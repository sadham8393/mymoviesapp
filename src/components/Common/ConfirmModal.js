import React from 'react';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import PropTypes from 'prop-types';

const ConfirmModal = (props) =>{
    return (
        <Modal centered isOpen={props.confirmModalOpen} className = "confirmModal">
            <ModalHeader>{props.title}</ModalHeader>
            <ModalBody>
                {props.confirmMessage}
            </ModalBody>
            <ModalFooter>
                <Button color="primary" onClick={props.onYesClick}>Yes</Button>
                <Button color="secondary" onClick={props.onNoClick}>No</Button>
            </ModalFooter>
        </Modal>
    );     
}

ConfirmModal.propTypes = { 
    confirmModalOpen: PropTypes.bool, 
    title:PropTypes.string,
    confirmMessage:PropTypes.string,
    onYesClick: PropTypes.func,
    onNoClick: PropTypes.func
}

export default ConfirmModal;