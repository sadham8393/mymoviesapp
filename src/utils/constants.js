// URl constants

export const MOVIEDB_PATH = "https://api.themoviedb.org/3/";
export const REACT_MOVIEDB_API_KEY  = "?api_key=0e89680c8b3fa0f259b53bbcc8f6091d";
export const MOVIEDB_IMAGE_APPEND_URL = "https://image.tmdb.org/t/p/w500/";
export const TRENDING_DAY_URL = "trending/all/day";


export const NO_IMAGE_URl = "https://s3-ap-southeast-1.amazonaws.com/upcode/static/default-image.jpg";

//Common Constants
export const INPUT_TYPE_TEXT = "text";
export const INPUT_TYPE_SELECT = "select";
export const NOT_AVAILABLE = "Not Available";


//Action Constants
export const FETCH_TRENDING_MOVIES = "FETCH_TRENDING_MOVIES";
export const FETCH_TV_MOVIES = "FETCH_TV_MOVIES";
export const DELETE_TV_MOVIES = "DELETE_TV_MOVIES";
export const MOVIES_FETCHED_SUCCESS = "MOVIES_FETCHED_SUCCESS";
export const MOVIES_FETCHED_ERROR = "MOVIES_FETCHED_ERROR";
export const DELETE_MOVIE_SUCCESS = "DELETE_MOVIE_SUCCESS";
export const DELETE_MOVIE_FAILURE = "DELETE_MOVIE_FAILURE";

export const MOVIES_FETCHED_NO_RESULTS = "MOVIES_FETCHED_NO_RESULTS";

export const RESET_RESPONSE = "RESET_RESPONSE";

export const NO_RESULTS_FOUND = "No results found for the given search input";

export const SERVER_UNAVAILABE = "Server unavailable. Please try after sometime"



