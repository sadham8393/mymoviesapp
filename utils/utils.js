import configureStore from 'redux-mock-store'

export const findByTestClass = (component, className) => {
    const wrapper = component.find(`${className}`);
    return wrapper;
};

export const findByComponent = (component, innerComponent) => {
    const wrapper = component.find(innerComponent);
    return wrapper;
};

export const testStore = (initialState) => {
    let store;
    const mockStore = configureStore();
    store = mockStore(initialState)
    return store;
};