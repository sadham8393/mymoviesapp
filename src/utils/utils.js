import { NOT_AVAILABLE } from './constants';

/**
 * Convert date format from "yyyy-mm-dd" to "dd/mm/yy"
 */

export function convertDate(date){
    if(date){
        return date.toString().split("-").reverse().join("/");
    }
    return NOT_AVAILABLE;
}