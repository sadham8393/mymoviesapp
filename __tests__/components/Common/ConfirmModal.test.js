import React from "react";
import { shallow, mount } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import { findByComponent, findByTestClass } from '../../../utils/utils';
import ConfirmModal  from "../../../src/components/Common/ConfirmModal";
import { Modal, ModalHeader, ModalBody, ModalFooter, Button} from 'reactstrap';

describe("ConfirmModal component", () => {
    describe('Renders correctly', () => {
        let wrapper;
        let mockFunc;
        let props;
        beforeEach(() => {
            mockFunc = jest.fn();
            props = {
                confirmModalOpen: true,
                title : "Avengers",
                confirmMessage : "Do you want to continue?",
                onYesClick : mockFunc,
                onNoClick : mockFunc
            };
            wrapper = shallow(<ConfirmModal {...props} />);
        });

        it('should render correctly', () => {
            const component = mount(<ConfirmModal {...props} />);
            expect(EnzymeToJson(component)).toMatchSnapshot();
        });

        it('should render Modal correctly and to be 1', () => {
            const modal = findByComponent(wrapper, Modal);
            expect(modal.length).toBe(1);
        });
    
        it('should render ModalHeader correctly and to be 1', () => {
            const modal = findByComponent(wrapper, ModalHeader);
            expect(modal.length).toBe(1);
        });
    
        it('should render ModalBody correctly and to be 1', () => {
            const modal = findByComponent(wrapper, ModalBody);
            expect(modal.length).toBe(1);
        });
    
        it('should render ModalFooter correctly and to be 1', () => {
            const modal = findByComponent(wrapper, ModalFooter);
            expect(modal.length).toBe(1);
        });

        it('should render Button correctly and to be 2', () => {
            const modal = findByComponent(wrapper, Button);
            expect(modal.length).toBe(2);
        });
    
        it('yes button should emit callback on click event', () => {
            const button = findByTestClass(wrapper, ".confirm-yes");
            button.simulate('click');
            const callback = mockFunc.mock.calls.length;
            expect(callback).toBe(1);
        });

        it('no button should emit callback on click event', () => {
            const button = findByTestClass(wrapper, ".confirm-no");
            button.simulate('click');
            const callback = mockFunc.mock.calls.length;
            expect(callback).toBe(1);
        });
    })
});