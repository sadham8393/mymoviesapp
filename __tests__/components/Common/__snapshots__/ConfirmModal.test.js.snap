// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal component Renders correctly should render correctly 1`] = `
<ConfirmModal
  confirmMessage="Do you want to continue?"
  confirmModalOpen={true}
  onNoClick={[MockFunction]}
  onYesClick={[MockFunction]}
  title="Avengers"
>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={true}
    className="confirmModal"
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    unmountOnClose={true}
    zIndex={1050}
  >
    <Portal
      node={
        <div
          style="position: relative; z-index: 1050; display: block;"
          tabindex="-1"
        >
          <div
            class=""
          >
            <div
              class="modal fade"
              role="dialog"
              style="display: block;"
              tabindex="-1"
            >
              <div
                class="modal-dialog confirmModal modal-dialog-centered"
                role="document"
              >
                <div
                  class="modal-content"
                >
                  <div
                    class="modal-header"
                  >
                    <h5
                      class="modal-title"
                    >
                      Avengers
                    </h5>
                  </div>
                  <div
                    class="modal-body"
                  >
                    Do you want to continue?
                  </div>
                  <div
                    class="modal-footer"
                  >
                    <button
                      class="confirm-yes btn btn-primary"
                      type="button"
                    >
                      Yes
                    </button>
                    <button
                      class="confirm-no btn btn-secondary"
                      type="button"
                    >
                      No
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="modal-backdrop fade"
            />
          </div>
        </div>
      }
    >
      <Portal
        containerInfo={
          <div
            style="position: relative; z-index: 1050; display: block;"
            tabindex="-1"
          >
            <div
              class=""
            >
              <div
                class="modal fade"
                role="dialog"
                style="display: block;"
                tabindex="-1"
              >
                <div
                  class="modal-dialog confirmModal modal-dialog-centered"
                  role="document"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      class="modal-header"
                    >
                      <h5
                        class="modal-title"
                      >
                        Avengers
                      </h5>
                    </div>
                    <div
                      class="modal-body"
                    >
                      Do you want to continue?
                    </div>
                    <div
                      class="modal-footer"
                    >
                      <button
                        class="confirm-yes btn btn-primary"
                        type="button"
                      >
                        Yes
                      </button>
                      <button
                        class="confirm-no btn btn-secondary"
                        type="button"
                      >
                        No
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="modal-backdrop fade"
              />
            </div>
          </div>
        }
      >
        <div
          className=""
        >
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onClick={[Function]}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            role="dialog"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex="-1"
            tag="div"
            timeout={300}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={300}
              unmountOnExit={false}
            >
              <div
                className="modal fade"
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                role="dialog"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className="modal-dialog confirmModal modal-dialog-centered"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <ModalHeader
                      charCode={215}
                      closeAriaLabel="Close"
                      tag="h5"
                      wrapTag="div"
                    >
                      <div
                        className="modal-header"
                      >
                        <h5
                          className="modal-title"
                        >
                          Avengers
                        </h5>
                      </div>
                    </ModalHeader>
                    <ModalBody
                      tag="div"
                    >
                      <div
                        className="modal-body"
                      >
                        Do you want to continue?
                      </div>
                    </ModalBody>
                    <ModalFooter
                      tag="div"
                    >
                      <div
                        className="modal-footer"
                      >
                        <Button
                          className="confirm-yes"
                          color="primary"
                          onClick={[MockFunction]}
                          tag="button"
                        >
                          <button
                            aria-label={null}
                            className="confirm-yes btn btn-primary"
                            onClick={[Function]}
                            type="button"
                          >
                            Yes
                          </button>
                        </Button>
                        <Button
                          className="confirm-no"
                          color="secondary"
                          onClick={[MockFunction]}
                          tag="button"
                        >
                          <button
                            aria-label={null}
                            className="confirm-no btn btn-secondary"
                            onClick={[Function]}
                            type="button"
                          >
                            No
                          </button>
                        </Button>
                      </div>
                    </ModalFooter>
                  </div>
                </div>
              </div>
            </Transition>
          </Fade>
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal-backdrop"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={150}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={150}
              unmountOnExit={false}
            >
              <div
                className="modal-backdrop fade"
              />
            </Transition>
          </Fade>
        </div>
      </Portal>
    </Portal>
  </Modal>
</ConfirmModal>
`;
